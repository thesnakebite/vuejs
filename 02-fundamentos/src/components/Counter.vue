<template>
    <div>
        <h2>{{ customTitle }}</h2>
        <p>{{ counter }} <sup>2</sup> = {{ squareCounter }}</p>

        <!-- <button v-on:click="getIncrease">+1</button> -->
        <button @click="getIncrease">+1</button>
        <button @click="getDecrease">-1</button>
    </div>
   
</template>

<script>
    export default{
        props: ['title'],

        name:'Counter',
         
        data() {
            return {
                counter: 5
            }
        },

        methods: {
            getSquareValue(){
                return this.counter * this.counter
            },

            // increase
            getIncrease(){
                this.counter++
            },

            // decrease
            getDecrease(){
                this.counter--
            }

        },

        computed: {
            squareCounter(){
                return this.counter * this.counter
            },

            customTitle(){
               return this.title || 'Counter'
            }
        }
    } 

</script>

<style scoped>
    button {
        background-color: #64BB87;
        border-radius: 5px;
        border: 1px solid greenyellow;
        color: white;
        cursor: pointer;
        margin: 0 5px;
        padding: 5px 15px;
        transition: 0.3s ease-in-out;
    }

    button:hover {
        background-color: #5aa67b;
        transform: 0.3s ease-in-out;
    }
</style>