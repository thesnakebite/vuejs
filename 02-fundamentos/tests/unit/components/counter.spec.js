import { shallowMount } from '@vue/test-utils'
import Counter from '@/components/Counter.vue'

describe('Counter Component', () => {
    test('Debe ser identico al image', () => {
        const wrapper = shallowMount( Counter )

        expect ( wrapper.html() ).toMatchSnapshot()

    })
})