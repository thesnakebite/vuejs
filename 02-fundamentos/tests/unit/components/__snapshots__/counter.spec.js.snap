// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Debe ser identico al image 1`] = `
<div class="buttons-container">
  <h2>Counter</h2>
  <p>10 <sup>2</sup> = 100</p><!-- <button v-on:click="getIncrease">+1</button> --><button>+1</button><button>-1</button>
</div>
`;
